pulsar-client/src/main/java/org/apache/pulsar/client/impl/TopicsConsumerImpl.java

```diff
@@ -726,26 +724,35 @@ private boolean topicNameValid(String topicName) {
     }
 
     // handling failure during subscribe new topic, unsubscribe success created partitions
-    private void handleSubscribeOneTopicError(String topicName, Throwable error) {
-        log.warn("[{}] Failed to subscribe for topic [{}] in topics consumer ", topic, topicName, error.getMessage());
+    private void handleSubscribeOneTopicError(String topicName, Throwable error, CompletableFuture<Void> subscribeFuture) {
+        log.warn("[{}] Failed to subscribe for topic [{}] in topics consumer {}", topic, topicName, error.getMessage());
+        topics.remove(topicName);
```

Comment:
any reason that topic is removed before closing the actual consumers?

Reply:
oh, Thanks, will move it into line 746, when all consumers closed.